datashape/discovery.py

```diff
@@ -405,3 +405,19 @@ def descendents(d, x):
         children -= desc
         desc.update(children)
     return desc
+
+
+Mock = None
+try:
+    from unittest.mock import Mock
+except ImportError:
+    try:
+        from mock import Mock
+    except ImportError:
+        pass
+
+if Mock is not None:
+    @dispatch(Mock)
```

Comment:
i know it's a little annoying but i think the proper solution is to define a catch-all simply raises rather than trying to catch numpy things and *then* raising. I'm happy to do this work, but it'll be at least a day before I can get to ti